worker/src/response.rs

```diff
@@ -306,33 +290,40 @@ impl Response {
     /// }
     /// ```
     pub fn with_cors(self, cors: &Cors) -> Result<Self> {
-        let mut headers = self.headers.clone();
+        let mut headers = self.init.headers.clone();
         cors.apply_headers(&mut headers)?;
         Ok(self.with_headers(headers))
     }
 
     /// Sets this response's `webSocket` option.
     /// This will require a status code 101 to work.
     pub fn with_websocket(mut self, websocket: Option<WebSocket>) -> Self {
-        self.websocket = websocket;
+        self.init.websocket = websocket;
         self
     }
 
     /// Read the `Headers` on this response.
     pub fn headers(&self) -> &Headers {
-        &self.headers
+        &self.init.headers
     }
 
     /// Get a mutable reference to the `Headers` on this response.
     pub fn headers_mut(&mut self) -> &mut Headers {
-        &mut self.headers
+        &mut self.init.headers
     }
 
     /// Clones the response so it can be used multiple times.
     pub fn cloned(&mut self) -> Result<Self> {
-        if self.websocket.is_some() {
+        if self.init.websocket.is_some() {
             return Err(Error::RustError("WebSockets cannot be cloned".into()));
         }
+        // This information is lost when we make the roundtrip though web_sys::Response
```

Comment:
Waiting to confirm if these properties are available on `Request`, which would allow for the round trip.